twitter4j-core/src/main/java/twitter4j/TwitterImpl.java

```diff
@@ -257,6 +264,114 @@ public UploadedMedia uploadMedia(String fileName, InputStream image) throws Twit
                 , new HttpParameter("media", fileName, image)).asJSONObject());
     }
 
+
+    @Override
+    public UploadedMedia uploadMediaChunked(File mediaFile) throws TwitterException {
+        checkFileValidity(mediaFile);
+        try {
+            return uploadMediaChunked(mediaFile.getName(), new FileInputStream(mediaFile), mediaFile.length());
+        } catch (IOException e) {
+            throw new TwitterException(e);
+        }
+    }
+
+
+	@Override
+	public UploadedMedia uploadMediaChunked(String fileName, InputStream media, long mediaLength) throws TwitterException {
+		try {
+			UploadedMedia uploadedMedia = uploadMediaChunkedInit(mediaLength);
+			BufferedInputStream buffered = new BufferedInputStream(media);
+
+			byte[] segmentData = new byte[chunkedUploadSegmentSize];
+			int totalRead = 0;
+
+			for (int bytesRead = 0, segmentIndex = 0; (bytesRead = buffered.read(segmentData)) > 0; ++segmentIndex) {
```

Comment:
`buffered.read(segmentData)` reads max 8K bytes only even with larger buffer size set during InputStream creation. This creates problem of more than 999 segment which breaks Twitter limit. And it creates a lot Append requests too.

Reply:
From my test, I saw most of the time max 8000 (not 8 * 1024) bytes will be read no matter what buffer size is used. And sometimes there can be some hundreds bytes.

Here is my implementation I am currently using. And this supports both gif and video upload.
https://github.com/eos1d3/twitter4j.  All codes based on dk8996 can't upload gif just because media category is wrong.